### PR TITLE
ICU-22807 Fix for missing __STDC_FORMAT_MACROS before inttypes.h

### PR DESCRIPTION
**Pull request by barracuda156**
This is a trivial fix for the issue which we have already addressed earlier. But now another instance of the same has to be fixed. See:
https://unicode-org.atlassian.net/browse/ICU-22807

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22807
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by richgillam — DISMISSED** [review]
Looks okay and matches an earlier fix that we all approved.

File: icu4c/source/i18n/messageformat2_function_registry.cpp [hunk 1]
```diff
@@ -19,6 +19,9 @@
 #include "number_types.h"
 #include "uvector.h" // U_ASSERT
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
```
Comment (markusicu) [inline icu4c/source/i18n/messageformat2_function_registry.cpp:24]: Would you mind using the formulation from PR #2250 ?
```suggestion
// The C99 standard suggested that C++ implementations not define PRId64 etc. constants
// unless this macro is defined.
// See the Notes at https://en.cppreference.com/w/cpp/types/integer .
// Similar to defining __STDC_LIMIT_MACROS in unicode/ptypes.h .
#ifndef __STDC_FORMAT_MACROS
#   define __STDC_FORMAT_MACROS
#endif
```

Comment (srl295) [reply icu4c/source/i18n/messageformat2_function_registry.cpp:24]: @barracuda156 ping?

Comment (barracuda156) [reply icu4c/source/i18n/messageformat2_function_registry.cpp:24]: @srl295 Sorry for a delay, will finalize this in a day or two.

**Comment by barracuda156** [comment]
@markusicu I have committed your suggestion. (Sorry, should have done that earlier…)

**Review by markusicu — APPROVED** [review]

**Comment by markusicu** [comment]
Thanks @barracuda156 !

**Pull request merged**